### DISCUSSION TITLE
Combining with schedulers, and question about input number

### DISCUSSION THREAD

**Discussion by jorlarsen** [erlendsellie/priceanalyzer#30] [General]
Hello.
Planning to use this thing with my existing automations. My existing automations does not care about the power prices, only about the time of day, and changes the set temperature of my floor heating based on this. (I have night, daytime and afternoon, and also checking for vacation mode).

My understanding is that the blueprint automation is adjusting the setpoint temperature value by either -1 og +1. So if my set temp is 26, and the price is about to get high, the automation noost the set temp to 27 for a while before the price goes high, correct?

If true, why do we need an input number helper, is it not possible to just alter the set temp directly with +-1?

**Comment by ArveVM** [comment]
Hi
Long time user of priceanalyzer here, but not particulary brilliant coder ;)

I'll try to explain in my own words; then Erlend for sure will follow up with some brilliant additional guiding if still required ;)

As I understand the concept:
- you need to have a virtual set-point that you can adjust. I call that my "LongTarget" - and it is the 'input number helper'  
- then comes along Priceanalyzer and gives indications of temp should be lower or higher to "buffer" or "save". I've visualized them as 'Priceanalyzer'  
- and then the sum of longterm and priceanalyzer is the target temp for the generic thermostat, 
- which is measured against the localTemp in the 'generic thermostat' which turns on/off heater according to settings in your longterm+priceanalyser vs local temp

<img width="500" alt="image" src="https://github.com/erlendsellie/priceanalyzer/assets/96014323/4eb5741c-2769-4380-9432-d8df5541020c">

PS: don't mind the powerauto that is part of a shell around to ensure I can override with following some local thermostats,,,

**Comment by erlendsellie** [comment]
Hi!
I completly forgot to answer this question - sorry about that.
It's like @ArveVM says, basically. I haven't found any way to hijack the thermostat-entity itself, so the thermostat setpoint will always be the final temperature.

It's the same with Tibber 'Smart varmestyring' as well, if you are familiar with it. You can configure it to control a Mill panel heater for example with the app, but if you then open the Mill app and override it, problems will arise. It's the same issue here - you must use the input_number to set the "setpoint" for priceanalyzer to work properly.

The upside of this is that the wife may turn the heater really high up, and then when priceananlyzer decides it's time to change the temperature again, it will be back to your preferred setpoint +/-  1(0) degree. So it will never be set at 28 degrees for a long period of time.

**Comment by jorlarsen** [comment]
Almost forgot about this myself. But thanks for the clarification around this case :)
